### PR TITLE
React Native TEST

### PR DESCRIPTION
**Pull request by Seb-AS**
React Native TEST

### CODE REVIEW

File: components/DrinkDetail.js [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+      this.state = {
+        visible: false,
+        loading: false,
+        data: [],
+        error: null,
+        refreshing: false
+      };
+    }
+   
+    componentDidMount() {
```
Comment (coreyakaplan) [inline components/DrinkDetail.js]: Please explain why you put this in componentDidMount. Is there a benefit or downside to doing it here?

File: components/DrinkDetail.js [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+        .then(res => {
+          this.setState({
+            data: res.drinks[0],
+            error: res.error || null,
+            loading: false,
+            refreshing: false
+          });
+        })
+        .catch(error => {
+          this.setState({ error, loading: false });
```
Comment (coreyakaplan) [inline components/DrinkDetail.js]: I see error in state, and being set here, however I don't see it being used anywhere? What would user interaction look like in this case, and how could you improve it?

File: components/DrinkDetail.js [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+        });
+    };
+  
+    render() {
+      const { data, isLoading } = this.state;
+
+      if (isLoading) {
+        return <p>Loading ...</p>;
```
Comment (coreyakaplan) [inline components/DrinkDetail.js]: This seems inconsistent. Why use ActivityIndicator in the other view, and a <p> here? Additionally, P is never actually declared anywhere. Would this compile?

File: components/DrinkDetail.js [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+          <View style={{ marginVertical: 10, flexDirection: 'row' }}>
+              <Image 
+                resizeMode='contain'
+                style={{
+                  flex: 1,
+                  aspectRatio: 1, 
+                  borderWidth: 1,
+                  borderColor: '#000',
+                  borderRadius: 5,
+                  height: 100,
```
Comment (coreyakaplan) [inline components/DrinkDetail.js]: Hard coded dimensions will not display properly on different device sizes. Please describe how you would handle styling for this app on both an ipad, and multiple sized iphones.

File: components/Home.js [hunk 5]
```diff
@@ -0,0 +1,172 @@
...
+
+  componentDidMount() {  
+    this.makeRemoteRequest();       
+  }
+
+  makeRemoteRequest = () => {
+    const url = "https://www.thecocktaildb.com/api/json/v1/1/filter.php?g=Cocktail_glass";
+    this.setState({ loading: true });
+
+    fetch(url)
```
Comment (coreyakaplan) [inline components/Home.js]: This is essentially the same code that we have in the details page. How could we consolidate it, and why would we want to?

File: components/DrinkDetail.js (hunk 2 above)
Comment (Seb-AS) [reply components/DrinkDetail.js]: render() {
    const { data, loading, error } = this.state;
  
    if (!loading) {
      if (error){
        return(
          <View>Please try again</View>
        );
      }
      return this.renderLoading();
    }

File: components/DrinkDetail.js (hunk 3 above)
Comment (Seb-AS) [reply components/DrinkDetail.js]: Sorry my own fault I was trying to be fast to complete the PR on time.


  render() {
    const { data, loading, error } = this.state;
  
    if (!loading) {
      if (error){
        return(
          <View>Please try again</View>
        );
      }
      return this.renderLoading();
    }

File: components/DrinkDetail.js (hunk 4 above)
Comment (Seb-AS) [reply components/DrinkDetail.js]: const {width, height} = require('Dimensions').get('window');
<Image 
              resizeMode='contain'
              style={{
                borderWidth: 1,
                borderColor: '#000',
                borderRadius: 5,
                width: width/2,
                height: width/2,
                borderRadius: 5
              }}
              prefetch={{ uri: strDrinkThumb }} 
              source={{ uri: strDrinkThumb }} 
            />

File: components/Home.js (hunk 5 above)
Comment (Seb-AS) [reply components/Home.js]: Sorry for the duplicated code.

I can do it dispatching payloads using redux or creating a Utils class and passing the url and getting a result object. and setting state.

http://www.thecocktaildb.com/api/json/v1/1/filter.php?g=Cocktail_glass
or 
http://www.thecocktaildb.com/api/json/v1/1/lookup.php?i=${idDrink} → Cocktail ID 

this.makeRemoteRequest(url);

  export const makeRemoteRequest = function (url) {
    fetch(url)
      .then((response) => response.json())
      .then((responseJson) => {
        return responseJson;
      })
      .catch((error) => {
        console.error(error);
        return null;(or the error)
    });

**Pull request closed without merging**